ethclient/ethclient.go

```diff
@@ -257,6 +257,14 @@ func toBlockNumArg(number *big.Int) string {
 	if number == nil {
 		return "latest"
 	}
+	switch number.Int64() {
+	case rpc.PendingBlockNumber.Int64():
+		return "pending"
+	case rpc.LatestBlockNumber.Int64():
+		return "latest"
+	case rpc.EarliestBlockNumber.Int64():
+		return "earliest"
+	}
```

Comment:
why is this needed? are you sure it has no side effect in code that use it and expect to get hex string?

Reply:
this is exactly the fix, json RPC can get either block number or one of these 3 tags. the API exposes just number as block number param, and I used those consts to define those special tags, eventually this code unmarshalled to json, and sent as json RPC.
you can see that the there is similar logic before my added switch, that send 'latest' when number is nil (default param is latest)

Reply:
cool 👏